apps/admin/frontend/src/api.ts

```diff
@@ -102,3 +104,243 @@ export const unprogramCard = {
     });
   },
 } as const;
+
+// Queries
+
+type QueryProps<T extends keyof Api> = Parameters<Api[T]>[0];
```

Comment:
While this works and is very cool, it's breaking the type abstraction of grout a bit (`Api` should be treated as an opaque type by consumers). The client types, however, are fair game to use, so how about something like this:

```ts
type QueryProps<Method extends keyof ApiClient> = Parameters<ApiClient[Method]>[0]
```

Reply:
If it works, it's good to me.